stress: rename Stress to DevStress

stevendanna opened the pull request:
This renames the current implementation of Stress to DevStress and then adds a new Stress function that covers both local and nightly stress runs.

Epic: none
Release note: None

stevendanna committed:
stress: rename Stress to DevStress

This renames the current implementation of Stress to DevStress and then
adds a new (unused) Stress function that covers both local and nightly
stress runs.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/154939)
<!-- Reviewable:end -->

jeffswenson reviewed (approved):
LGTM

I think the skip methods should check the new `Stress` helper, but this is a clarity improvement even without that change.

stevendanna committed:
skip: UnderStress* should skip all stress runs

We previously thought that the fact that skip.UnderStress has been
ineffective in engflow that it was OK to change the behaviour to only
apply to dev builds. But this is not the case for the Extended CI
pipeline which still runs pre-merge.

Epic: none
Release note: None
